src/hdmf/container.py

```diff
@@ -746,11 +747,11 @@ def __smart_str_dict(d, num_indent):
         out += '\n' + indent + right_br
         return out
 
-    def set_data_io(self, dataset_name, data_io_class, **kwargs):
+    def set_data_io(self, dataset_name: str, data_io_class: Type[DataIO], data_io_kwargs: dict):
```

Comment:
This change is also inspired for consistency; usually the `**kwargs` usage across HDMF refers to the ones that get popped and passed on by docval

Reply:
a bit of a backwards compatibility break though. Could you do a a soft deprecation?

Reply:
yes, I see how this could be confusing. Can we try to do a soft deprecation, since this technically breaks backwards compatibility? Maybe something like 

```
def set_data_io(self, dataset_name: str, data_io_class: Type[DataIO], data_io_kwargs: dict, **kwargs):
    if kwargs:
        DeprecationWarning("Please pass the kwargs in `data_io_kwargs` instead")
        data_io_kwargs = kwargs
    ...
```